Compile error for apex/csrc/mlp.cpp : invalid narrowing conversion from "unsigned long" to "signed long"

torch version 1.10.2 with cu113
nvcc info
```
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Mon_May__3_19:15:13_PDT_2021
Cuda compilation tools, release 11.3, V11.3.109
Build cuda_11.3.r11.3/compiler.29920130_0
```

I try to install Nvidia-apex with command below

```
pip install -v --disable-pip-version-check --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
```
then I got error message below 

```2022-02-06T01:14:48,482   /home/dngusdnr1/apex/csrc/mlp.cpp(57): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,482       for (int i = 0; i < num_layers; i++) {
2022-02-06T01:14:48,482                         ^

2022-02-06T01:14:48,482   /home/dngusdnr1/apex/csrc/mlp.cpp(65): error #3291: invalid narrowing conversion from "unsigned long" to "signed long"
2022-02-06T01:14:48,482       auto reserved_space = at::empty({reserved_size}, inputs[0].type());
2022-02-06T01:14:48,482                                        ^

2022-02-06T01:14:48,483   /home/dngusdnr1/apex/csrc/mlp.cpp(69): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,483       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_forward", [&] {
2022-02-06T01:14:48,483       ^

2022-02-06T01:14:48,483   /home/dngusdnr1/apex/csrc/mlp.cpp(69): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,483       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_forward", [&] {
2022-02-06T01:14:48,483       ^

2022-02-06T01:14:48,484   /home/dngusdnr1/apex/csrc/mlp.cpp(69): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,484       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_forward", [&] {
2022-02-06T01:14:48,484       ^

2022-02-06T01:14:48,484   /home/dngusdnr1/apex/csrc/mlp.cpp(115): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,484       for (int i = 0; i < num_layers; i++) {
2022-02-06T01:14:48,485                         ^

2022-02-06T01:14:48,485   /home/dngusdnr1/apex/csrc/mlp.cpp(120): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,485       for (int i = 0; i < inputs.size(); i++) {
2022-02-06T01:14:48,485                         ^

2022-02-06T01:14:48,485   /home/dngusdnr1/apex/csrc/mlp.cpp(124): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,485       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,486       ^

2022-02-06T01:14:48,486   /home/dngusdnr1/apex/csrc/mlp.cpp(124): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,486       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,486       ^

2022-02-06T01:14:48,486   /home/dngusdnr1/apex/csrc/mlp.cpp(124): error #3291: invalid narrowing conversion from "unsigned long" to "signed long"
2022-02-06T01:14:48,487       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,487       ^

2022-02-06T01:14:48,487   /home/dngusdnr1/apex/csrc/mlp.cpp(124): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,487       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,487       ^

2022-02-06T01:14:48,488   /home/dngusdnr1/apex/csrc/mlp.cpp(124): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,488       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,488       ^

2022-02-06T01:14:48,488   /home/dngusdnr1/apex/csrc/mlp.cpp(124): error #3291: invalid narrowing conversion from "unsigned long" to "signed long"
2022-02-06T01:14:48,488       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,488       ^

2022-02-06T01:14:48,489   /home/dngusdnr1/apex/csrc/mlp.cpp(124): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,489       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,489       ^

2022-02-06T01:14:48,489   /home/dngusdnr1/apex/csrc/mlp.cpp(124): warning #2557: comparison between signed and unsigned operands
2022-02-06T01:14:48,489       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,489       ^

2022-02-06T01:14:48,490   /home/dngusdnr1/apex/csrc/mlp.cpp(124): error #3291: invalid narrowing conversion from "unsigned long" to "signed long"
2022-02-06T01:14:48,490       AT_DISPATCH_FLOATING_TYPES_AND_HALF(inputs[0].type(), "mlp_backward", [&] {
2022-02-06T01:14:48,490       ^

2022-02-06T01:14:48,490   /home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include/torch/ordered_dict.h(360): warning #1011: missing return statement at end of non-void function "torch::OrderedDict<Key, Value>::operator[](const Key &)"
2022-02-06T01:14:48,490     }
2022-02-06T01:14:48,491     ^

2022-02-06T01:14:48,491   /home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include/torch/ordered_dict.h(368): warning #1011: missing return statement at end of non-void function "torch::OrderedDict<Key, Value>::operator[](const Key &) const"
2022-02-06T01:14:48,491     }
2022-02-06T01:14:48,491     ^

2022-02-06T01:14:48,491   compilation aborted for /home/dngusdnr1/apex/csrc/mlp.cpp (code 2)
2022-02-06T01:15:54,298   [2/2] /home/dngusdnr1/anaconda3/envs/loop_se3/bin/nvcc  -I/home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/include -I/home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include -I/home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/include/TH -I/home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/include/THC -I/home/dngusdnr1/anaconda3/envs/loop_se3/include -I/home/dngusdnr1/anaconda3/envs/loop_se3/include/python3.9 -c -c /home/dngusdnr1/apex/csrc/mlp_cuda.cu -o /home/dngusdnr1/apex/build/temp.linux-x86_64-3.9/csrc/mlp_cuda.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O3 -ccbin=/usr/bin/g++ -DVERSION_GE_1_1 -DVERSION_GE_1_3 -DVERSION_GE_1_5 --threads 4 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=mlp_cuda -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_60,code=sm_60 -gencode=arch=compute_61,code=sm_61 -gencode=arch=compute_62,code=sm_62 -gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_86,code=sm_86 -std=c++14
2022-02-06T01:15:54,299   ninja: build stopped: subcommand failed.
2022-02-06T01:15:54,300   Traceback (most recent call last):
2022-02-06T01:15:54,300     File "/home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 1717, in _run_ninja_build
2022-02-06T01:15:54,301       subprocess.run(
2022-02-06T01:15:54,301     File "/home/dngusdnr1/anaconda3/envs/loop_se3/lib/python3.9/subprocess.py", line 528, in run
2022-02-06T01:15:54,302       raise CalledProcessError(retcode, process.args,
2022-02-06T01:15:54,302   subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.
```

Used compiler version is 
``` /usr/bin/g++ --version
g++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Please help me. Thanks in advances.

--- a/csrc/mlp.cpp
+++ b/csrc/mlp.cpp
@@ -62,7 +62,7 @@ std::vector<at::Tensor> mlp_forward(int use_bias, int activation, std::vector<at
 
   // create output/workspace tensor
   auto out = at::empty({batch_size, output_features.back()}, inputs[0].type());
-  auto reserved_space = at::empty({reserved_size}, inputs[0].type());
+  auto reserved_space = at::empty({static_cast<long>(reserved_size)}, inputs[0].type());
   // allocate fixed 4MB workspace for cublaslt for now, and this gets at least 4 MB
   auto lt_workspace = at::empty({1 << 22}, inputs[0].type());
 
@@ -135,7 +135,7 @@ std::vector<at::Tensor> mlp_backward(
         get_mlp_bp_workspace_in_bytes<scalar_t>(batch_size, num_layers, output_features.data());
 
     // auto work_space = at::empty({work_size*4}, at::kByte);
-    auto work_space = at::empty({work_size / sizeof(scalar_t)}, inputs[0].type());
+    auto work_space = at::empty({static_cast<long>(work_size / sizeof(scalar_t))}, inputs[0].type());
 
     auto result = mlp_bp<scalar_t>(
         inputs[0].data_ptr<scalar_t>(),

